service/topology_coordinator.cc

```diff
@@ -752,6 +753,87 @@ class topology_coordinator : public endpoint_lifecycle_subscriber {
         case global_topology_request::cleanup:
             co_await start_cleanup_on_dirty_nodes(std::move(guard), true);
             break;
+        case global_topology_request::keyspace_rf_change: {
+            while (true) {
+                auto tmptr = get_token_metadata_ptr();
+                sstring ks_name = *_topo_sm._topology.new_keyspace_rf_change_ks_name;
+                std::unordered_map<sstring, sstring> new_rf_per_dc = *_topo_sm._topology.new_keyspace_rf_change_rf_per_dc;
+                std::map<sstring, sstring> new_rf_per_dc_map{new_rf_per_dc.begin(), new_rf_per_dc.end()};
+                sstring req_uuid = *_topo_sm._topology.global_request_id;
+                std::vector<canonical_mutation> updates;
+                sstring error;
+                auto& ks = _db.find_keyspace(ks_name);
+                tablet_replica_calculation_result new_data{locator::tablet_map{8}, tablet_replica_calculation_status::success};
+
+                for (const auto& table : ks.metadata()->tables()) {
+                    auto repl_strategy = ks.get_replication_strategy_ptr()->maybe_as_tablet_aware();
+                    try {
+                        new_data = co_await service::calculate_tablet_replicas_for_new_rf(
+                                repl_strategy, table, tmptr, new_rf_per_dc_map);
+                    } catch(const std::exception& e) {
+                        new_data.status = tablet_replica_calculation_status::other;
+                        error = e.what();
+                        rtlogger.error("{}", error);
+                    }
+
+                    if (new_data.status != tablet_replica_calculation_status::success) {
+                        rtlogger.error("Couldn't process global_topology_request::keyspace_rf_change, status: {}, rf per dc map: {}",
+                                        new_data.status, new_data.new_tablet_map);
+                        updates.clear(); // remove all tablets mutations ...
+                        break;           // ... and only create mutations deleting the global req
```

Comment:
This seems quite fragile with the reset of the code being oblivious to the error.
How would that be maintained in the future?
I think that a better approach would be to wrap the suffix functionality in a function with a documented interface, if possible, and call it from both the good path and the error path.

Reply:
Code has changed since this comment has been written, does it still apply?
The remaining part of the code acts as both the good and bad path, there's only 1 difference between the two and it's checking `error` to see what exactly to execute.